packages/hooks/useFetchSVG/useFetchSVG.tsx

```diff
@@ -0,0 +1,40 @@
+import { h, VNode } from 'preact';
+import { useEffect, useState } from 'preact/hooks';
+
+const convertDocToSVGElem = (node: HTMLElement): VNode | string | null | undefined => {
+  if (node.nodeType === Node.TEXT_NODE) return node.nodeValue;
+  if (node.nodeType === Node.ELEMENT_NODE) {
+    const children = (Array.from(node.childNodes) as HTMLElement[]).map(convertDocToSVGElem);
+    const props = Object.fromEntries(
+      Array.from(node.attributes).map((attr) => [attr.nodeName, attr.nodeValue]),
+    );
+    return h(node.nodeName, props, children);
+  }
+  return null;
+};
+
+export const useFetchSVG = (name: string) => {
+  const [svgComponent, setSVGComponent] = useState<string | VNode | null | undefined>(null);
+
+  useEffect(() => {
+    const fetchSVG = async () => {
+      try {
+        const response = await fetch(
+          `https://firebasestorage.googleapis.com/v0/b/no-gravity-76bb2.appspot.com/o/icons%2F${name}.svg?alt=media`,
+        );
+        const svgText = await response.text();
+        const domParser = new DOMParser();
+        const svgDoc = domParser.parseFromString(svgText, 'image/svg+xml');
+        const svgElem = convertDocToSVGElem(svgDoc.documentElement);
+        setSVGComponent(svgElem);
+      } catch (error) {
+        // eslint-disable-next-line
+        console.error('Error fetching SVG:', error);
+      }
+    };
```

Comment:
I think we should use a modern data fetching library and leverage the caching here. It could really reduce the ammount of requests we need to make to firebase. Could also improve load speeds if that is ever an issue.

Let's have a talk!

Reply:
Or at the very least make our own little caching implementation